logback/src/main/java/com/linecorp/armeria/common/logback/RequestContextExportingAppender.java

```diff
@@ -229,6 +238,9 @@ public void stop() {
 
     @Override
     public void addAppender(Appender<ILoggingEvent> newAppender) {
+        if (newAppender instanceof AbstractSocketAppender) {
+            needsHashMap = true;
```

Comment:
Could you leave a comment that we discussed why we use the `HashMap` and link to the whitelist check?

Reply:
done